### PR TITLE
feat: Added the merchant name as Suggestion in expense detail card for CC expenses

### PR DESCRIPTION
**Pull request by devendrafyle**
## Clickup
https://app.clickup.com/t/86cxbrpvr

## UI Preview
image
image
image

## Summary by CodeRabbit

- **New Features**
	- Added merchant information display for corporate credit card transactions, including a modal for more details.
	- Introduced a new popover component for displaying merchant information.

- **Bug Fixes**
	- Improved error handling for mandatory fields in expense forms.

- **Style**
	- Enhanced styling for merchant-related elements and transaction status display.

- **Tests**
	- Added test cases for the new popover component and its integration within existing pages.

### CODE REVIEW

File: src/app/shared/components/cc-expense-merchant-info-popover/cc-expense-merchant-info-popover.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Component } from '@angular/core';
+import { PopoverController } from '@ionic/angular';
+
+@Component({
+  selector: 'app-cc-expense-merchant-info',
+  templateUrl: './cc-expense-merchant-info-popover.component.html',
+  styleUrls: ['./cc-expense-merchant-info-popover.component.scss'],
+})
+export class CcExpenseMerchantInfoPopoverComponent {
```
Comment (Dimple16) [inline src/app/shared/components/cc-expense-merchant-info-popover/cc-expense-merchant-info-popover.component.ts:9]: update the name `CCExpenseMerchantInfoPopoverComponent`

**Review by Dimple16 — APPROVED** [review]
minor comment

File: src/app/shared/components/cc-expense-merchant-info-popover/cc-expense-merchant-info-popover.component.ts (hunk 1 above)
Comment (devendrafyle) [reply src/app/shared/components/cc-expense-merchant-info-popover/cc-expense-merchant-info-popover.component.ts:9]: Sure

File: src/app/fyle/view-expense/view-expense.page.html [hunk 2]
```diff
@@ -100,6 +100,21 @@
...
         </div>
 
+        <ng-container *ngIf="isCCCTransaction">
+          <div class="view-expense--card-container__merchant_block">
+            <div class="view-expense--card-container__merchant_label">Merchant:</div>
+            <div class="view-expense--card-container__merchant_value">
+              <span> {{ selectedCCCTransaction?.vendor || 'Unavailable' }} </span>
+              <ion-icon
+                (click)="(openCCExpenseMerchantInfoModal())"
+                src="../../../assets/svg/info-circle-outline.svg"
```
[the remark this answers is not in the document]
Comment (Chethan-Fyle) [reply src/app/fyle/view-expense/view-expense.page.html:110]: +1

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**